docs/dev/allocation.qmd

```diff
@@ -0,0 +1,279 @@
+---
+title: "Allocation"
+---
+# Overview of allocation implementation
+In this document, the allocation workflow is explained. Below is an overview of it.
+```{mermaid}
+flowchart TD
+    subgraph update_allocation
+        direction TB
+        G[update mean flows]-->E[collect demand]
+        E-->F[allocate]
+    end
+    style update_allocation fill:#9ff
+    C(Begin)-->A[Initialization allocation]
+    A--> update_allocation --> H[\end of time?\]
+    H--> |No| update_allocation
+    H --> |Yes| D(End)
+```
+
+If allocation is used in a model, [Allocation structs](https://deltares.github.io/Ribasim/dev/allocation.html#the-allocation-struct) are created.
+The allocation struct stores the data that is needed for the calculations and stores also the results of the calculation.
+In allocation, optimization is an essential part.
+`JuMP.jl` is used to model and solve the optimization problems that are defined by allocation.
+The [AllocationModel struct](https://deltares.github.io/Ribasim/dev/allocation.html#the-allocation-struct) is used for constructing the JuMP model.
+When an instance of `AllocationModel` is created, a JuMP optimization model is defined and initialized in the instance.
+More details on how allocation interacts with `JuMP.jl` is explained [here](https://deltares.github.io/Ribasim/dev/allocation.html#jump.jl-problem-interaction).
+
+After initialization, as the simulation starts, the allocation problem is solved and updated after every allocation timestep (which is specified in the TOML).
+With every allocation timestep a new optimization problem is formulated and solved, using the latest available (simulation) model conditions and forcing and demand predictions.
+
+The update of allocation (`update_allocation`) is repeating and spread into three parts:
+
+- Updating the mean flows. The mean flow data is only used only for output, not used by any internal functions.
+- ["Collect demand"](https://deltares.github.io/Ribasim/concept/allocation.html#the-high-level-algorithm). This step initialize and solve the optimization problems that collects the demand from the subnetworks.
+- ["Allocate"](https://deltares.github.io/Ribasim/concept/allocation.html#the-high-level-algorithm). This step solves the optimization problems that allocates the demand to the subnetworks.
```

Comment:
For the main network this step allocates to the subnetworks (and demand nodes that are in the main network). For the subnetwork this step allocates to the demand nodes.